src/backend/access/transam/xlog.c

```diff
@@ -8142,6 +8142,8 @@ StartupXLOG(void)
 	 */
 	if (fast_promoted)
 		RequestCheckpoint(CHECKPOINT_FORCE);
+
+	*shmCleanupBackends = true;
```

Comment:
> 1. Why have you removed the first assignment and left only the last?

Because at the end of the function, the assignment will still work, and this is also necessary for their commented out test.

Reply:
> 2\. `shmCleanupBackends` will be set to `true` in both cases

What are the two cases? There is only one condition in the patch!

Reply:
> 2\. will this variable always be true on QD?

Yes, that seems to be the case.